operator/pkg/controller/istiocontrolplane/istiocontrolplane_controller.go

```diff
@@ -288,7 +288,6 @@ func (r *ReconcileIstioOperator) Reconcile(_ context.Context, request reconcile.
 				return reconcile.Result{}, nil
 			} else if errors.IsConflict(finalizerError) {
 				scope.Infof("Could not remove finalizer from %s due to conflict. Operation will be retried in next reconcile attempt.", iopName)
-				return reconcile.Result{}, nil
```

Comment:
I would suggest to replace `return reconcile.Result{}, nil` with `continue, so that the `finalizerMaxRetries` still works. If you remove `return reconcile.Result{}, nil`, it will be forced to next reconcile attempt.

Reply:
let me look at that

Reply:
@morvencao that codepoint is not in a loop so cannot do a continue.
I can return with the error immediately it occurs rather than let it drop through?
Is that ok?

Reply:
then the `finalizerMaxRetries` will take no effect, right?

Reply:
@morvencao apologies had misunderstood your original comment.

Really I only need the retry in the create side, should not have made change in the remove part.

LEt me resubmit

Reply:
AM I right in saying that the maxRetries is only concerned with conflict error in the finalizer removal, not the addition?

For my use case I really am concerned only with the create case.

I can revert the change in the remove part.

Reply:
So the change mean that you want to force the controller to start new reconcile loop once finalizer add conflict is found.

Reply:
@morvencao 

yes - otherwise a temporary conflict will never be resolved. The issue here occurs when delivering the operator profile by helm3.

Helm3 operator delivers the operator profile doc and adds a label to it `app.kubernetes.io/managed-by: Helm`. But there is a race as the resource state has now changed whillst the istio operator reconcile logic is running

Meanwhile istio-operator is trying to process causing an IOP error (which is not logged - I got this log by allowing the error to drop trhough:

```
2022-04-27T12:15:43.253727Z     info    installer       Reconciling IstioOperator
2022-04-27T12:15:43.858528Z     info    Adding finalizer istio-finalizer.install.istio.io to istio-system/istio-control-plane
2022-04-27T12:15:43.869335Z     info    installer       Could not add finalizer to istio-control-plane due to conflict. Operation will be retried in next reconcile attempt.
2022-04-27T12:15:43.869376Z     error   installer       Failed to add finalizer to IstioOperator CR istio-control-plane: Operation cannot be fulfilled on istiooperators.install.istio.io "istio-control-plane": the object has been modified; please apply your changes to the latest version and try again    moreInfo=A finalizer could not be added to the IstioOperator resource. The controller uses the finalizer to temporarily prevent the resource from being deleted while the Istio control plane is being deleted. impact=When the IstioOperator resource is deleted, the Istio control plane may not be fully removed. action=Delete and re-add the IstioOperator resource. If this error persists, see https://istio.io/latest/about/bugs for possible solutions. likelyCause=The likely cause is a problem with the Kubernetes API server.
```

As it stands without this change there is no subsequent attempt made to resolve this and the processing of the operator resource is never retried.

There are other possibilities for a transient  conflict to occur and the reconcile loop should resolve this by re-running